openssl-sys/build/main.rs

```diff
@@ -226,6 +226,11 @@ See rust-openssl documentation for more information:
         }
     }
 
+    for enabled in &enabled {
+        println!("cargo:rustc-cfg=osslconf=\"{}\"", enabled);
+    }
+    println!("cargo:conf={}", enabled.join(","));
+
```

Comment:
I think if you move the contents of `check_ssl_kind` to right here, you'll get what you want, functionally.